### ISSUE TITLE
SDL2 Compilation Error on Linux Wayland

### ISSUE THREAD

**Issue by desttinghim** [nesbox/TIC-80#1769]
When trying to build TIC80 master on Arch Linux with sway, everything seems to go fine until linking player-sdl:

```
[100%] Linking CXX executable bin/player-sdl
/usr/bin/ld: lib/libSDL2.a(SDL_waylandvideo.c.o): in function `wl_display_get_registry':
/home/desttinghim/repos/TIC-80/build/vendor/sdl2/wayland-generated-protocols/wayland-client-protocol.h:1032: undefined reference to `wl_proxy_marshal_flags'
/usr/bin/ld: lib/libSDL2.a(SDL_waylandvideo.c.o): in function `wl_registry_bind':
/home/desttinghim/repos/TIC-80/build/vendor/sdl2/wayland-generated-protocols/wayland-client-protocol.h:1145: undefined reference to `wl_proxy_marshal_flags'
/usr/bin/ld: /home/desttinghim/repos/TIC-80/build/vendor/sdl2/wayland-generated-protocols/wayland-client-protocol.h:1145: undefined reference to `wl_proxy_marshal_flags'
/usr/bin/ld: /home/desttinghim/repos/TIC-80/build/vendor/sdl2/wayland-generated-protocols/wayland-client-protocol.h:1145: undefined reference to `wl_proxy_marshal_flags'
/usr/bin/ld: /home/desttinghim/repos/TIC-80/build/vendor/sdl2/wayland-generated-protocols/wayland-client-protocol.h:1145: undefined reference to `wl_proxy_marshal_flags'
/usr/bin/ld: lib/libSDL2.a(SDL_waylandvideo.c.o):/home/desttinghim/repos/TIC-80/build/vendor/sdl2/wayland-generated-protocols/wayland-client-protocol.h:1145: more undefined references to `wl_proxy_marshal_flags' follow
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/player-sdl.dir/build.make:108: bin/player-sdl] Error 1
make[1]: *** [CMakeFiles/Makefile2:419: CMakeFiles/player-sdl.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```
It looks like this is an issue with SDL2, although it looks (https://github.com/libsdl-org/SDL/issues/5088) like it's been fixed upstream?

**Comment by nesbox** [comment]
Will be fixed with the next SDL2 lib update.
Thank you.

**Comment by silverhikari** [comment]
still getting this error after the [commit](https://github.com/libsdl-org/SDL/commit/e2ade2bfc46d915cd306c63c830b81d800b2575f) which was added in sdl2 release 2.0.20

**Comment by nesbox** [comment]
It's bad, I don't know how to fix it :(

**Comment by messju** [comment]
I think the problem is not the sdl2-2.0.20 on arch linux. The Problem seems to be the the git submodule in vendor/sdl2 is used and it is pinned to 2.0.14. I not that deep in git submodules and cmake, but

> git submodule update --remote vendor/sdl2

fixed that for me temporarily.

**Comment by nesbox** [comment]
Fixed with #1825
